### DISCUSSION TITLE
Style

### DISCUSSION THREAD

**Discussion by paulushub** [opengeospatial/geoapi#106] [Q&A] [answered]
Just for clarifications. 
The current style package is still pending, but will not compile due to outdated `Expression` and `Filter` interfaces.
I know the next iteration of the symbolizers specification is due for approval in 2029, but what is the current plan
of the GeoAPI with regards to the styles package?

**Comment by desruisseaux** [comment, marked as answer]
The `org.opengis.style` package is not part of the GeoAPI 3.0 release. Officially, it should be as if it does not exist. The Geotk project uses it, but the fact that Geotk is using a non-released version of GeoAPI is a problem that we are trying to resolve. On Geotk side, we are considering to replace `org.opengis.style` usages by `org.apache.sis.style.se1` internally (the `se1` part remind that this is a model derived from an encoding). The content of the `org.opengis.style` package may change completely, and it may be safer to delete it until a new style specification is ready at OGC. But there is no formal decision (except that `org.opengis.style` is not part of official release), so its fate is still open to discussion.

**Comment by desruisseaux** [comment]
However, the `org.opengis.style` package should still compile. It compiles on my side. What are the compilation errors that you get?

**Reply by paulushub** [reply]
Not even tried to compile, **eclipse** is already displaying error. With the generic expressions defined as
```java
@UML(identifier="Expression", specification=ISO_19143)
public interface Expression<R,V> extends Function<R,V> {
}
```
I was not expecting the following to compile
```java
@XmlElement("AnchorPoint")
public interface AnchorPoint {
    /**
     * get the x coordinate of the anchor point
     *
     * @return the expression which represents the X coordinate
     */
    @XmlElement("AnchorPointX")
    Expression getAnchorPointX();

    /**
     * get the y coordinate of the anchor point
     *
     * @return the expression which represents the Y coordinate
     */
    @XmlElement("AnchorPointY")
    Expression getAnchorPointY();

    /**
     * calls the visit method of a StyleVisitor
     *
     * @param visitor the style visitor
     */
    Object accept(StyleVisitor visitor, Object extraData);
}
```

**Reply by paulushub** [reply]
Sorry: Just tried it in NetBeans and it compiles with Java 21. Java Generic is really strange!

**Reply by desruisseaux** [reply]
Thanks. But Eclipse was also correct, above `Expression` usages, while compilable, are incorrect. The message reported by Eclipse was probably a warning rather than an error. This is part of the reasons why this package needs to be revisited.
